### ISSUE TITLE
all-last fungerar inte som förut/som det ska

### ISSUE THREAD

**Issue by sebbe34-svt** [spaam/svtplay-dl#1662]
Jag har för mig denna 'feature/bug' introducerades vid en annan bugrättning, så jag är försiktig med att rapportera denna, då jag inte vill att tidigare bug ska återuppstå?

Men när man har en sida med flera avsnitt, kan man t.ex. använda argument -A, med argument -R för att välja från vilket 'håll' den ska börja ladda ner, vilket ju är bra om man har bråttom och inte vill ladda ner alla avsnitt, t.ex.. (då kan man ju avbryta manuellt när man är nöjd). 
Men om man vill 'hoppa över' NN antal avsnitt därifrån man börjar (framifrån eller bakifrån med -R) med argumentet --all-last, så görs inte detta, utan det börjar ladda ner från början ändå, och gissar jag, hoppas över de sista avsnitten istället! Vilket känns lite meningslöst (detta kan man ju avbryta manuellt själv) - viktigare är att kunna hoppa över några avsnitt därifrån man börjar, för att ladda ner "de sista" istället (t.e.x när man har bråttom, eller redan laddat ner de första avsnitten..)

Jag *hoppas* att min förklaring inte blev alltför svår att förstå :) - poängen är iaf med argumentet --all-last ,  att man vill ju kunna hoppa över X antal avsnitt i början, när man börjar ladda ner en serie.. (oavsett om det är framifrån eller bakifrån med argument -R)
/mvh Sebbe

**Comment by spaam** [comment]
vilken sida testar du detta på  när de inte fungerar? tv4play?

**Comment by sebbe34-svt** [comment]
jag kör på svtplay just nu (t.ex. https://www.svtplay.se/video/eorz4qk/stromso/jultrojan-i-ny-tappning-sy-en-julvast-av-gamla-julgardiner
)

**Comment by spaam** [comment]
jag tror jag förstår va du försöker säga(?).

tex om serien har avsnitt 1 2 3 4 5. du har tankat hem 1 och 2 redan. sedan vill du skippa dom två avsnitten och börja tanka hem 3, 4 och 5 nästa gång? 

om så , just nu skippar den dom två avsnitten (1 och 2) automagiskt om du tankar hem de i samma katalog (om du nu inte har bytt namn på filerna, då funkar det inte) och sedan fortsätter med 3, 4 och 5.  men om man har `--all-last 3` så kommer den välja 1 ,2 och 3.  men iom 1 och 2 redan är nere så kommer den bara tanka hem nummer 3. 

```
svtplay-dl https://www.svtplay.se/video/eorz4qk/stromso/jultrojan-i-ny-tappning-sy-en-julvast-av-gamla-julgardiner -A -R --all-last 3 
INFO: Episode 1 of 3
INFO: Url: http://www.svtplay.se/video/eorz4qk/stromso/jultrojan-i-ny-tappning-sy-en-julvast-av-gamla-julgardiner
INFO: Outfile: stromso.jultrojan.i.ny.tappning-sy.en.julvast.av.gamla.julgardiner-5adab73-svtplay.ts
WARNING: File (stromso.jultrojan.i.ny.tappning-sy.en.julvast.av.gamla.julgardiner-5adab73-svtplay.ts) already exists. Use --force to overwrite
INFO: Episode 2 of 3
INFO: Url: http://www.svtplay.se/video/jb312Wv/stromso/gor-egna-facklor-av-konserver-och-hudkram-i-fast-form
INFO: Outfile: stromso.gor.egna.facklor.av.konserver.och.hudkram.i.fast.form-822d606-svtplay.ts
INFO: Selected to download hls, bitrate: 3376 format: h264
[047/444][===========
```

men men just nu finns inget argument just nu för att skippa X avsnitt manuellt då ingen har frågat om det o så. 🤔

**Comment by sebbe34-svt** [comment]
Jja, det var så jag menade. Och det jag efterfrågar var så det funkade innan, x antal versioner tillbaka. (issue #1613 påminner om denna, vet ej om OP hade samma sak i åtanke; version 4.71 misstänker jag var versionen då det slutade fungera.)

Hm, jag tänker också att det jag efterfrågar är en viktigare och mer flexible funktionalitet: i en väldigt lång lista med avsnitt (t.ex. med många säsonger) vill man kunna hoppa över en säsong t.ex., men fortsatta på nästa säsong och ladda ner alla avsnitt därifrån. (man har inte alltid lyxen att ladda ner alla tidigare avsnitt först)

Den funktionalitet som finns nu ser jag inte riktigt vitsen med..? att sluta ladda ner efter x avsnitt? om det laddas ner för många kan man ju alltid ta bort dem efteråt?

**Comment by sebbe34-svt** [comment]
Kan tillägga också, att detta är störst problem på tv4play, för där kan man inte, som jag förstår det, längre ladda ner enstaka avsnitt på en sida? Man måste ladda ner allt som finns på sidan, (även om det är flera säsonger.)
 (.. eller, som --all-last fungerar nu, kan man förstås begränsa till att ladda ner X avsnitt från början/slutet, men då går det alltså inte att "komma" åt enbart avsnitten däremellan längre.. vilket gick förut... :(

**Comment by spaam** [comment]
klart man kan ladda hem ett enskilt program på tv4play.  funkade nyss för mig. har du testat på enskild video utan -A ?  låter som Kamrat har inte gjort det.

**Comment by sebbe34-svt** [comment]
Hm? även om jag ej använder -A på en "serie-sida", så laddas alla avsnitt ner..
ex:
svtplay-dl -q 2628 https://www.tv4play.se/program/3d92dfaa8978372b42d7/jul-med-ernst
->
svtplay-dl -q 2628 https://www.tv4play.se/program/3d92dfaa8978372b42d7/jul-med-ernst
INFO: Episode 1 of 47
INFO: Url: https://www.tv4play.se/video/9ad8d418a137fb34fb23
..
etc...

Och försöker jag ta en länk för ett enskilt avsnitt istället, får jag felmeddelande:

ex: $ svtplay-dl https://www.tv4play.se/video/7e492a4137e8469b08f3/avsnitt-1-locka-igang-julen
WARNING: Use program page instead of the video one.

Så har det varit länge.. men inte innan autenticeringsmetoden ändrades.
(för info: svtplay-dl --version
svtplay-dl 4.101

**Comment by spaam** [comment]
låter som kamrat har en [config](https://svtplay-dl.se/config/) fil där kamrat har sagt att den ska använda `all_episodes: true` för tv4play alternativt alla sidor iom att du inte har specifierat `-A` argumentet. 

som det ska se ut: 
```
fallen ~ » cat ~/.svtpaly-dl.yaml
service:
  svtplay:
    format_preferred: h264,hevc,h264-51,hevc-51

  tv4play:
    token: "eyJhb....."
fallen ~ » svtplay-dl https://www.tv4play.se/video/7e492a4137e8469b08f3/avsnitt-1-locka-igang-julen
INFO: Outfile: jul.med.ernst.s16e01.locka.igang.julen-7e492a4137e8469b08f3-tv4play.ts
INFO: Selected to download hls, bitrate: 7027 format: h264
......
```

med `all_episodes` för tv4play: 
```
fallen ~ » cat ~/.svtpaly-dl.yaml
service:
  svtplay:
    format_preferred: h264,hevc,h264-51,hevc-51

  tv4play:
    all_episodes: true
    token: "eyJhb....."
fallen ~ » svtplay-dl https://www.tv4play.se/video/7e492a4137e8469b08f3/avsnitt-1-locka-igang-julen
WARNING: Use program page instead of the video one.
```

med `all_episodes` för allt:
```
default:
  all_episodes: true

service:
  svtplay:
    format_preferred: h264,hevc,h264-51,hevc-51

  tv4play:
    token: "eyJhb....."

fallen ~ » svtplay-dl https://www.tv4play.se/video/7e492a4137e8469b08f3/avsnitt-1-locka-igang-julen
WARNING: Use program page instead of the video one.

```

**Comment by spaam** [comment]
Om du nu ska visa din config fil. snälla ta bort din token så folk slipper se den.  den är knuten till ditt konto och är en bra grej att inte dela med sig utav.

**Comment by sebbe34-svt** [comment]
Ja, så var det tydligen! Tack för tipset! Det underlättar ju en hel del! :)  
(Glömt hur detta smög sig in där..  men felmeddelandet kunde kanske varit tydligare.. att man valt -A men att detta inte funkar på enskilt avsnitt för tv4... då hade jag kunnat gissa att min config-fil hade med det att göra)

(Men för att få ännu mer kontroll på 'automatiken' (och tydligheten)   föreslår jag att all-last ändrar namn till skip-last, och/eller ännu hellre att man hade en till parameter också som hette skip-first ... då hade man med stor precision kunnat sikta sig in på ett viss antal avsnitt, även om de ligger "mitt i" i  listan på sidan.)
